crates/bevy_text/src/text.rs

```diff
@@ -165,41 +165,48 @@ impl TextLayout {
 
 /// A span of text in a tree of spans.
 ///
-/// `TextSpan` is only valid as a child of an entity with [`TextLayout`], which is provided by `Text`
-/// for text in `bevy_ui` or `Text2d` for text in 2d world-space.
-///
+/// `TextSpan` is only valid as a child of an entity with `Text` or `Text2d` which is the first span.
```

Comment:
Unfortunately, it's hard to clearly convey this thought without ambiguities. How about something like:

---

A `TextSpan` is only valid when it exists as a child entity of a parent that has either a `Text` or `Text2d` component.

The parent entity's `Text`/`Text2d` component contains the base text content. Any child entities with `TextSpan` components extend this text by appending their content to the parent's text in sequence.